Приветсвуем всех любителей тру крайма!
Сегодня мы будем анализировать базу данных "Отчеты об убийствах, 1980-2014 гг. в США"

Проект подотчетности за убийство является самой полной базой данных об убийствах
в Соединенных Штатах, доступной в настоящее время. Этот набор данных включает в себя убийства 
из Дополнительного отчета ФБР об убийствах с 1976 года по настоящее время и данные 
Закона о свободе информации о более чем 22 000 убийствах, о которых не было сообщено в Министерство юстиции.
Этот набор данных включает в себя возраст, расу, пол, этническую принадлежность жертв и преступников,
в дополнение к отношениям между жертвой и преступником и используемым оружием.



In [42]:
import pandas as pd
import numpy as np

In [43]:
data= pd.read_csv('database.csv')

C:\Users\molde\AppData\Local\Temp\ipykernel_34432\1511261187.py:1: DtypeWarning: Columns (16) have mixed types. Specify dtype option on import or set low_memory=False.
  data= pd.read_csv('database.csv')


In [3]:
data.head()

Record ID Agency Code Agency Name       Agency Type       City   State  \
0          1     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
1          2     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
2          3     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
3          4     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
4          5     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   

   Year    Month  Incident              Crime Type  ... Victim Ethnicity  \
0  1980  January         1  Murder or Manslaughter  ...          Unknown   
1  1980    March         1  Murder or Manslaughter  ...          Unknown   
2  1980    March         2  Murder or Manslaughter  ...          Unknown   
3  1980    April         1  Murder or Manslaughter  ...          Unknown   
4  1980    April         2  Murder or Manslaughter  ...          Unknown   

  Perpetrator Sex  Perpetrator Age               Perpetrator Race  \
0            Male               15  Native American/Alaska Native   
1            Male               42                          White   
2         Unknown                0                        Unknown   
3            Male               42                          White   
4         Unknown                0                        Unknown   

  Perpetrator Ethnicity  Relationship         Weapon Victim Count  \
0               Unknown  Acquaintance   Blunt Object            0   
1               Unknown  Acquaintance  Strangulation            0   
2               Unknown       Unknown        Unknown            0   
3               Unknown  Acquaintance  Strangulation            0   
4               Unknown       Unknown        Unknown            0   

  Perpetrator Count Record Source  
0                 0           FBI  
1                 0           FBI  
2                 0           FBI  
3                 0           FBI  
4                 1           FBI  

[5 rows x 24 columns]

In [4]:
data.shape

(638454, 24)

Дата фрейм состоит из 24 столбцов и 638454 строк

In [5]:
## проведем анализ исходных данных, определив при этом тип данных в исходном датафрейме.
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 638454 entries, 0 to 638453
Data columns (total 24 columns):
 #   Column                 Non-Null Count   Dtype 
---  ------                 --------------   ----- 
 0   Record ID              638454 non-null  int64 
 1   Agency Code            638454 non-null  object
 2   Agency Name            638454 non-null  object
 3   Agency Type            638454 non-null  object
 4   City                   638454 non-null  object
 5   State                  638454 non-null  object
 6   Year                   638454 non-null  int64 
 7   Month                  638454 non-null  object
 8   Incident               638454 non-null  int64 
 9   Crime Type             638454 non-null  object
 10  Crime Solved           638454 non-null  object
 11  Victim Sex             638454 non-null  object
 12  Victim Age             638454 non-null  int64 
 13  Victim Race            638454 non-null  object
 14  Victim Ethnicity       638454 non-null  object
 15  

Датасет содержит 7 столбцов формата int 64 и 18 столбцов формата object

Проанализируем данные при помощи оператора describe для категорийных признаков. Мы видим общее количество признаков поо строкам 'count', количество уникальных значений 'unique', наименование наиболее часто повторяющегося 'top', и частоту повторения этого признака 'freq'. 

In [6]:
data.describe(include='object')

Agency Code Agency Name       Agency Type         City       State  \
count       638454      638454            638454       638454      638454   
unique       12003        9216                 7         1782          51   
top        NY03030    New York  Municipal Police  Los Angeles  California   
freq         38416       38416            493026        44511       99783   

         Month              Crime Type Crime Solved Victim Sex Victim Race  \
count   638454                  638454       638454     638454      638454   
unique      12                       2            2          3           5   
top       July  Murder or Manslaughter          Yes       Male       White   
freq     58696                  629338       448172     494125      317422   

       Victim Ethnicity Perpetrator Sex  Perpetrator Age Perpetrator Race  \
count            638454          638454           638454           638454   
unique                3               3              191                5   
top             Unknown            Male                0            White   
freq             368303          399541           211079           218243   

       Perpetrator Ethnicity Relationship   Weapon Record Source  
count                 638454       638454   638454        638454  
unique                     3           28       16             2  
top                  Unknown      Unknown  Handgun           FBI  
freq                  446410       273013   317484        616647

In [7]:
data.describe()

Record ID           Year       Incident     Victim Age  \
count  638454.00000  638454.000000  638454.000000  638454.000000   
mean   319227.50000    1995.801102      22.967924      35.033512   
std    184305.93872       9.927693      92.149821      41.628306   
min         1.00000    1980.000000       0.000000       0.000000   
25%    159614.25000    1987.000000       1.000000      22.000000   
50%    319227.50000    1995.000000       2.000000      30.000000   
75%    478840.75000    2004.000000      10.000000      42.000000   
max    638454.00000    2014.000000     999.000000     998.000000   

        Victim Count  Perpetrator Count  
count  638454.000000      638454.000000  
mean        0.123334           0.185224  
std         0.537733           0.585496  
min         0.000000           0.000000  
25%         0.000000           0.000000  
50%         0.000000           0.000000  
75%         0.000000           0.000000  
max        10.000000          10.000000

Анализ числовых даннных стандартен. Видим общее количество заначений признаков, и основноые статистические величины (среднее арифметическое, стандарное отклонение, минимальное и макисмальное значения а также квантили 25,50,75)

In [8]:
# мы изменили тип данных, а также поставили нули там, где возраст - пустая строка, чтобы показать, что значения нет (пропуск)
def int_(x):
    try:
        return int(x)
    except:
        return 0
data['Perpetrator Age']=data['Perpetrator Age'].apply(int_)

# Фильтрация 

1. Сколько же было несовершеннолетних преступников, то есть младше 18? Убираем нули, так как это неизвестные возрасты преступников.

In [9]:
data[(data['Perpetrator Age'] < 18) &(data['Perpetrator Age'] != 0)]

Record ID Agency Code          Agency Name       Agency Type  \
0               1     AK00101            Anchorage  Municipal Police   
31             32     AKASP00  Alaska State Police      State Police   
35             36     AKASP00  Alaska State Police      State Police   
60             61     AL00100            Jefferson           Sheriff   
86             87     AL00102           Birmingham  Municipal Police   
...           ...         ...                  ...               ...   
638355     638356     WIMPD00            Milwaukee  Municipal Police   
638360     638361     WIMPD00            Milwaukee  Municipal Police   
638370     638371     WV00500        Brooke County           Sheriff   
638441     638442     WY00300      Campbell County           Sheriff   
638443     638444     WY01101             Cheyenne  Municipal Police   

             City          State  Year     Month  Incident  \
0       Anchorage         Alaska  1980   January         1   
31         Juneau         Alaska  1980      June         1   
35         Juneau         Alaska  1980    August         1   
60      Jefferson        Alabama  1980  November         1   
86      Jefferson        Alabama  1980  February         3   
...           ...            ...   ...       ...       ...   
638355  Milwaukee      Wisconsin  2014  December         1   
638360  Milwaukee      Wisconsin  2014  December         6   
638370     Brooke  West Virginia  2014       May         1   
638441   Campbell        Wyoming  2014  February         1   
638443    Laramie        Wyoming  2014   October         1   

                        Crime Type  ... Victim Ethnicity Perpetrator Sex  \
0           Murder or Manslaughter  ...          Unknown            Male   
31      Manslaughter by Negligence  ...          Unknown            Male   
35          Murder or Manslaughter  ...          Unknown            Male   
60      Manslaughter by Negligence  ...     Not Hispanic            Male   
86          Murder or Manslaughter  ...          Unknown          Female   
...                            ...  ...              ...             ...   
638355      Murder or Manslaughter  ...          Unknown            Male   
638360      Murder or Manslaughter  ...          Unknown            Male   
638370      Murder or Manslaughter  ...          Unknown          Female   
638441      Murder or Manslaughter  ...     Not Hispanic            Male   
638443      Murder or Manslaughter  ...          Unknown            Male   

        Perpetrator Age               Perpetrator Race Perpetrator Ethnicity  \
0                    15  Native American/Alaska Native               Unknown   
31                    7         Asian/Pacific Islander               Unknown   
35                   17  Native American/Alaska Native               Unknown   
60                   15                          White          Not Hispanic   
86                   17                          Black               Unknown   
...                 ...                            ...                   ...   
638355               17                          Black               Unknown   
638360               17                          Black               Unknown   
638370               16                          White               Unknown   
638441               15                          White          Not Hispanic   
638443               16  Native American/Alaska Native               Unknown   

        Relationship        Weapon Victim Count Perpetrator Count  \
0       Acquaintance  Blunt Object            0                 0   
31            Sister         Rifle            0                 0   
35           Unknown         Rifle            0                 0   
60            Friend       Shotgun            0                 0   
86      Acquaintance       Shotgun            0                 0   
...              ...           ...          ...               ...   
638355      Stranger       Handgun       

In [10]:
round((data[(data['Perpetrator Age'] < 18) &(data['Perpetrator Age'] != 0)].shape[0] / data.shape[0] *100),2)

5.32

5,32% преступлений (то есть 33967 из 638454) совершили несовершенноление, младше 18 лет.

2. Интересно посмотреть, против кого совершались эти преступления: против несовершеннолетних или более старших людей?

In [11]:
data[(data['Perpetrator Age'] < 18) &(data['Perpetrator Age'] != 0) & (data['Victim Age'] < 18)]

Record ID Agency Code          Agency Name       Agency Type  \
0               1     AK00101            Anchorage  Municipal Police   
31             32     AKASP00  Alaska State Police      State Police   
60             61     AL00100            Jefferson           Sheriff   
129           130     AL00102           Birmingham  Municipal Police   
150           151     AL00102           Birmingham  Municipal Police   
...           ...         ...                  ...               ...   
638122     638123     WA03105           Marysville  Municipal Police   
638123     638124     WA03105           Marysville  Municipal Police   
638337     638338     WIMPD00            Milwaukee  Municipal Police   
638355     638356     WIMPD00            Milwaukee  Municipal Police   
638360     638361     WIMPD00            Milwaukee  Municipal Police   

             City       State  Year      Month  Incident  \
0       Anchorage      Alaska  1980    January         1   
31         Juneau      Alaska  1980       June         1   
60      Jefferson     Alabama  1980   November         1   
129     Jefferson     Alabama  1980     August         6   
150     Jefferson     Alabama  1980   November         2   
...           ...         ...   ...        ...       ...   
638122  Snohomish  Washington  2014    October         1   
638123  Snohomish  Washington  2014   November         1   
638337  Milwaukee   Wisconsin  2014  September         1   
638355  Milwaukee   Wisconsin  2014   December         1   
638360  Milwaukee   Wisconsin  2014   December         6   

                        Crime Type  ... Victim Ethnicity Perpetrator Sex  \
0           Murder or Manslaughter  ...          Unknown            Male   
31      Manslaughter by Negligence  ...          Unknown            Male   
60      Manslaughter by Negligence  ...     Not Hispanic            Male   
129         Murder or Manslaughter  ...     Not Hispanic            Male   
150         Murder or Manslaughter  ...     Not Hispanic            Male   
...                            ...  ...              ...             ...   
638122      Murder or Manslaughter  ...     Not Hispanic            Male   
638123      Murder or Manslaughter  ...     Not Hispanic            Male   
638337      Murder or Manslaughter  ...          Unknown            Male   
638355      Murder or Manslaughter  ...          Unknown            Male   
638360      Murder or Manslaughter  ...          Unknown            Male   

        Perpetrator Age               Perpetrator Race Perpetrator Ethnicity  \
0                    15  Native American/Alaska Native               Unknown   
31                    7         Asian/Pacific Islander               Unknown   
60                   15                          White          Not Hispanic   
129                  17                          White          Not Hispanic   
150                  16                          Black          Not Hispanic   
...                 ...                            ...                   ...   
638122               15  Native American/Alaska Native          Not Hispanic   
638123               15  Native American/Alaska Native          Not Hispanic   
638337               16                          Black               Unknown   
638355               17                          Black               Unknown   
638360               17                          Black               Unknown   

        Relationship        Weapon Victim Count Perpetrator Count  \
0       Acquaintance  Blunt Object            0                 0   
31            Sister         Rifle            0                 0   
60            Friend       Shotgun            0                 0   
129     Acquaintance       Shotgun            0                 0   
150     Acquaintance       Handgun            0                 0   
...              ...           ...          ...               ...   
638122        Friend       Handgun            2                 0

In [12]:
round ((data[(data['Perpetrator Age'] < 18) &(data['Perpetrator Age'] != 0) & (data['Victim Age'] < 18)].shape[0]  / data[(data['Perpetrator Age'] < 18) &(data['Perpetrator Age'] != 0)].shape[0]*100),2)

31.59

31,59 % преступлений из тех, которые были совершены преступниками младше 18,и направлены против несовершеннолетних. Получается, что оставшиеся 68,41 % преступлений были направлены на более старших жертв. Видим интересную закономерность: несовершеннолетние чаще всего совершают преступления против старших

3. Самый интересный вопрос. Кто убивает чаще: мужчины или женщины? Отчаянные домохозяйки или агенты 007? 

In [13]:
data['Perpetrator Sex'].unique()

array(['Male', 'Unknown', 'Female'], dtype=object)

In [14]:
data[data['Perpetrator Sex']=='Male']

Record ID Agency Code      Agency Name       Agency Type       City  \
0               1     AK00101        Anchorage  Municipal Police  Anchorage   
1               2     AK00101        Anchorage  Municipal Police  Anchorage   
3               4     AK00101        Anchorage  Municipal Police  Anchorage   
5               6     AK00101        Anchorage  Municipal Police  Anchorage   
6               7     AK00101        Anchorage  Municipal Police  Anchorage   
...           ...         ...              ...               ...        ...   
638446     638447     WY01300   Natrona County           Sheriff    Natrona   
638447     638448     WY01301           Casper  Municipal Police    Natrona   
638448     638449     WY01301           Casper  Municipal Police    Natrona   
638450     638451     WY01700  Sheridan County           Sheriff   Sheridan   
638452     638453     WY01800  Sublette County           Sheriff   Sublette   

          State  Year      Month  Incident              Crime Type  ...  \
0        Alaska  1980    January         1  Murder or Manslaughter  ...   
1        Alaska  1980      March         1  Murder or Manslaughter  ...   
3        Alaska  1980      April         1  Murder or Manslaughter  ...   
5        Alaska  1980        May         1  Murder or Manslaughter  ...   
6        Alaska  1980        May         2  Murder or Manslaughter  ...   
...         ...   ...        ...       ...                     ...  ...   
638446  Wyoming  2014  September         1  Murder or Manslaughter  ...   
638447  Wyoming  2014       June         1  Murder or Manslaughter  ...   
638448  Wyoming  2014     August         1  Murder or Manslaughter  ...   
638450  Wyoming  2014       June         1  Murder or Manslaughter  ...   
638452  Wyoming  2014   December         1  Murder or Manslaughter  ...   

       Victim Ethnicity Perpetrator Sex  Perpetrator Age  \
0               Unknown            Male               15   
1               Unknown            Male               42   
3               Unknown            Male               42   
5               Unknown            Male               36   
6               Unknown            Male               27   
...                 ...             ...              ...   
638446     Not Hispanic            Male               48   
638447     Not Hispanic            Male               22   
638448          Unknown            Male               67   
638450          Unknown            Male               57   
638452     Not Hispanic            Male               31   

                     Perpetrator Race Perpetrator Ethnicity  Relationship  \
0       Native American/Alaska Native               Unknown  Acquaintance   
1                               White               Unknown  Acquaintance   
3                               White               Unknown  Acquaintance   
5                               White               Unknown  Acquaintance   
6                               Black               Unknown          Wife   
...                               ...                   ...           ...   
638446                          White          Not Hispanic        Mother   
638447                          White               Unknown    Girlfriend   
638448                          Black               Unknown        Friend   
638450                          White               Unknown  Acquaintance   
638452                          White          Not Hispanic      Stranger   

               Weapon Victim Count Perpetrator Count Record Source  
0        Blunt Object            0                 0           FBI  
1       Strangulation            0                 0           FBI  
3       Strangulation            0                 0           FBI  
5               Rifle            0                 0           FBI  
6               Knife            0                 0           FBI  
...               ...          ...               ...           ...  
638446        Handgun        

In [15]:
data[data['Perpetrator Sex']=='Female']

Record ID Agency Code               Agency Name       Agency Type  \
15             16     AK00101                 Anchorage  Municipal Police   
27             28     AKASP00       Alaska State Police      State Police   
28             29     AKASP00       Alaska State Police      State Police   
46             47     AKASP00       Alaska State Police      State Police   
48             49     AL00100                 Jefferson           Sheriff   
...           ...         ...                       ...               ...   
638433     638434     WVWSP53  State Police: Upperglade      State Police   
638434     638435     WVWSP55       State Police: Welch      State Police   
638435     638436     WVWSP55       State Police: Welch      State Police   
638451     638452     WY01701                  Sheridan  Municipal Police   
638453     638454     WY01902              Rock Springs  Municipal Police   

              City          State  Year      Month  Incident  \
15       Anchorage         Alaska  1980     August         3   
27          Juneau         Alaska  1980   February         2   
28          Juneau         Alaska  1980   February         3   
46          Juneau         Alaska  1980   December         2   
48       Jefferson        Alabama  1980      March         1   
...            ...            ...   ...        ...       ...   
638433     Webster  West Virginia  2014      March         1   
638434    McDowell  West Virginia  2014    October         1   
638435    McDowell  West Virginia  2014    October         1   
638451    Sheridan        Wyoming  2014  September         1   
638453  Sweetwater        Wyoming  2014  September         1   

                        Crime Type  ... Victim Ethnicity Perpetrator Sex  \
15          Murder or Manslaughter  ...          Unknown          Female   
27          Murder or Manslaughter  ...          Unknown          Female   
28          Murder or Manslaughter  ...          Unknown          Female   
46      Manslaughter by Negligence  ...          Unknown          Female   
48          Murder or Manslaughter  ...          Unknown          Female   
...                            ...  ...              ...             ...   
638433      Murder or Manslaughter  ...          Unknown          Female   
638434      Murder or Manslaughter  ...          Unknown          Female   
638435      Murder or Manslaughter  ...          Unknown          Female   
638451      Murder or Manslaughter  ...          Unknown          Female   
638453      Murder or Manslaughter  ...     Not Hispanic          Female   

        Perpetrator Age               Perpetrator Race Perpetrator Ethnicity  \
15                   29                          Black               Unknown   
27                   28  Native American/Alaska Native               Unknown   
28                   36  Native American/Alaska Native               Unknown   
46                   27  Native American/Alaska Native               Unknown   
48                   38                          White               Unknown   
...                 ...                            ...                   ...   
638433               30                          White               Unknown   
638434               35                          White               Unknown   
638435               35                          White               Unknown   
638451               22         Asian/Pacific Islander               Unknown   
638453               24                          White          Not Hispanic   

       Relationship        Weapon Victim Count Perpetrator Count Record Source  
15       Ex-Husband       Handgun            0                 0           FBI  
27          Husband       Handgun            0                 0           FBI  
28          Brother         Rifle            0                 0           FBI  
46          Husband         Rifle            0                 0           FBI  
48          Husband       Shotgun  

In [16]:
round ((data[data['Perpetrator Sex']=='Female'].shape[0] / data.shape[0] *100) ,2)

7.6

In [17]:
round ((data[data['Perpetrator Sex']=='Male'].shape[0] / data.shape[0] *100) ,2)

62.58

По общим данным всех годов получается, что мужчины совершают преступления почти в 8 раз чаще, чем женщины (женщины совершили 48548 преступлений - 7.6%, а мужчины - 399541 - 62.58% преступлений )

4. А если пол неизвестен, получается, что преступление не раскрыто?

In [18]:
round ((data[(data['Perpetrator Sex']=='Unknown')&(data['Crime Solved']=='No')].shape[0] / 
 data[(data['Perpetrator Sex']=='Unknown')].shape[0] *100),2)

99.92

Более 99.9% преступлений, совершенных неизвестным полом, не раскрыты. Мы оказались правы. 

In [19]:
data[(data['Perpetrator Sex']=='Unknown')&(data['Crime Solved']=='Yes')]

Record ID Agency Code      Agency Name       Agency Type        City  \
405788     405789     FL01300       Miami-Dade     County Police        Dade   
405796     405797     FL01300       Miami-Dade     County Police        Dade   
405972     405973     FL01602     Jacksonville  Municipal Police       Duval   
406255     406256     FL04800    Orange County           Sheriff      Orange   
406259     406260     FL04800    Orange County           Sheriff      Orange   
...           ...         ...              ...               ...         ...   
627925     627926     FL05027     Palm Springs  Municipal Police  Palm Beach   
627977     627978     FL05214   St. Petersburg  Municipal Police    Pinellas   
628035     628036     FL05601      Fort Pierce  Municipal Police   St. Lucie   
628040     628041     FL05601      Fort Pierce  Municipal Police   St. Lucie   
628063     628064     FL05800  Sarasota County           Sheriff    Sarasota   

          State  Year      Month  Incident              Crime Type  ...  \
405788  Florida  2000       July        55  Murder or Manslaughter  ...   
405796  Florida  2000     August        63  Murder or Manslaughter  ...   
405972  Florida  2000    January       239  Murder or Manslaughter  ...   
406255  Florida  2000  September       522  Murder or Manslaughter  ...   
406259  Florida  2000    October       526  Murder or Manslaughter  ...   
...         ...   ...        ...       ...                     ...  ...   
627925  Florida  2014    October       978  Murder or Manslaughter  ...   
627977  Florida  2014    January        30  Murder or Manslaughter  ...   
628035  Florida  2014    January        88  Murder or Manslaughter  ...   
628040  Florida  2014      April        93  Murder or Manslaughter  ...   
628063  Florida  2014   December       116  Murder or Manslaughter  ...   

       Victim Ethnicity Perpetrator Sex  Perpetrator Age Perpetrator Race  \
405788          Unknown         Unknown                0          Unknown   
405796          Unknown         Unknown                0          Unknown   
405972          Unknown         Unknown                0          Unknown   
406255          Unknown         Unknown                0          Unknown   
406259          Unknown         Unknown                0          Unknown   
...                 ...             ...              ...              ...   
627925          Unknown         Unknown                0          Unknown   
627977          Unknown         Unknown                0          Unknown   
628035          Unknown         Unknown                0          Unknown   
628040          Unknown         Unknown                0          Unknown   
628063          Unknown         Unknown                0          Unknown   

       Perpetrator Ethnicity Relationship        Weapon Victim Count  \
405788               Unknown      Unknown       Handgun            1   
405796               Unknown      Unknown       Handgun            1   
405972               Unknown      Unknown       Handgun            1   
406255               Unknown      Unknown       Handgun            1   
406259               Unknown      Unknown       Unknown            1   
...                      ...          ...           ...          ...   
627925               Unknown      Unknown  Blunt Object            1   
627977               Unknown      Unknown         Knife            1   
628035               Unknown      Unknown       Handgun            1   
628040               Unknown      Unknown       Handgun            1   
628063               Unknown      Unknown       Handgun            1   

       Perpetrator Count Record Source  
405788                 0          FOIA  
405796                 0          FOIA  
405972                 0          FOIA  
406255                 0          FOIA  
406259                 0          FOIA  
...                  ...           ...  
627925                 0          FOIA  
627977                 0    

Оставшийся 0.01% раскрытых преступлений имеют, видимо, закрытые данные, так как видим, что многие поля имеют статус Unknown 

5. В какой год преступления совершались чаще всего? 

In [20]:
data['Year'].mode()

0    1993
Name: Year, dtype: int64

Чаще всего преступления совершались в 1993 году

In [21]:
data[data['Year']==1993]

Record ID Agency Code   Agency Name       Agency Type        City  \
266227     266228     AK00101     Anchorage  Municipal Police   Anchorage   
266228     266229     AK00101     Anchorage  Municipal Police   Anchorage   
266229     266230     AK00101     Anchorage  Municipal Police   Anchorage   
266230     266231     AK00101     Anchorage  Municipal Police   Anchorage   
266231     266232     AK00101     Anchorage  Municipal Police   Anchorage   
...           ...         ...           ...               ...         ...   
290557     290558     WY01600        Platte           Sheriff      Platte   
290558     290559     WY01800      Sublette           Sheriff    Sublette   
290559     290560     WY01900    Sweetwater           Sheriff  Sweetwater   
290560     290561     WY01902  Rock Springs  Municipal Police  Sweetwater   
290561     290562     WY02101      Evanston  Municipal Police       Uinta   

          State  Year      Month  Incident                  Crime Type  ...  \
266227   Alaska  1993        May         2      Murder or Manslaughter  ...   
266228   Alaska  1993    January         1      Murder or Manslaughter  ...   
266229   Alaska  1993   February         1      Murder or Manslaughter  ...   
266230   Alaska  1993   February         2  Manslaughter by Negligence  ...   
266231   Alaska  1993      March         1      Murder or Manslaughter  ...   
...         ...   ...        ...       ...                         ...  ...   
290557  Wyoming  1993        May         1      Murder or Manslaughter  ...   
290558  Wyoming  1993  September         1      Murder or Manslaughter  ...   
290559  Wyoming  1993    January         1      Murder or Manslaughter  ...   
290560  Wyoming  1993      March         1      Murder or Manslaughter  ...   
290561  Wyoming  1993     August         1      Murder or Manslaughter  ...   

       Victim Ethnicity Perpetrator Sex  Perpetrator Age  \
266227          Unknown            Male               20   
266228          Unknown         Unknown                0   
266229          Unknown          Female               39   
266230          Unknown          Female               20   
266231          Unknown         Unknown                0   
...                 ...             ...              ...   
290557          Unknown          Female               19   
290558          Unknown         Unknown                0   
290559          Unknown            Male               28   
290560          Unknown          Female               20   
290561          Unknown            Male               21   

                     Perpetrator Race Perpetrator Ethnicity Relationship  \
266227                          Black               Unknown       Sister   
266228                        Unknown               Unknown      Unknown   
266229                          White               Unknown    Boyfriend   
266230  Native American/Alaska Native               Unknown          Son   
266231                        Unknown               Unknown      Unknown   
...                               ...                   ...          ...   
290557                          White               Unknown    Boyfriend   
290558                        Unknown               Unknown      Unknown   
290559                          White               Unknown     Stranger   
290560                          White               Unknown     Daughter   
290561                          White               Unknown      Ex-Wife   

              Weapon Victim Count Perpetrator Count Record Source  
266227       Handgun            1                 0           FBI  
266228       Handgun            0                 0           FBI  
266229         Knife            0                 0           FBI  
266230      Drowning            0                 0           FBI  
266231         Knife            0                 0           FBI  
...              ...          ...               ...           ...  
290557         Rifle    

In [22]:
round ((data[data['Year']==1993].shape[0] /data.shape[0]*100),2)

3.81

24335 преступлений были совершены в самый "популярный" год в наших данных. Это почти 4% от всех данных. 

6. А  в каких штатах в 1993 году В "популярном" 1993 году чаще всего  совершались преступления? Выведим топ 5. 

In [23]:
data[data['Year']==1993]['State'].value_counts().head()

State
California    4369
New York      2511
Texas         2271
Florida       1329
Illinois      1013
Name: count, dtype: int64

В "популярном" 1993 году преступления совершались в разных штатах, в топ-5 попали: Калифорния, Нью-Йорк, Техас, Флорида и Иллинойс. 

In [24]:
data[data['Year']==1993]['State'].value_counts().tail()

State
South Dakota    21
Wyoming         18
Vermont         13
North Dakota    11
Maine            7
Name: count, dtype: int64

А это топ-5 'наоборот' (наименьшее количество)

7. Кто чаще совершал преступления в 1993 году? 

In [25]:
data[data['Year']==1993]['Perpetrator Sex'].value_counts()

Perpetrator Sex
Male       14842
Unknown     7900
Female      1593
Name: count, dtype: int64

В "популярном" 1993 году преступления совершали чаще всего мужчины, в 14 раз меньше - женщины

8. А против кого чаще всего совершались преступления? 

In [26]:
data[(data['Year']==1993) & (data['Victim Sex']=='Female')]

Record ID Agency Code   Agency Name       Agency Type        City  \
266227     266228     AK00101     Anchorage  Municipal Police   Anchorage   
266235     266236     AK00101     Anchorage  Municipal Police   Anchorage   
266237     266238     AK00101     Anchorage  Municipal Police   Anchorage   
266238     266239     AK00101     Anchorage  Municipal Police   Anchorage   
266247     266248     AK00101     Anchorage  Municipal Police   Anchorage   
...           ...         ...           ...               ...         ...   
290553     290554     WY01200       Lincoln           Sheriff     Lincoln   
290555     290556     WY01301        Casper  Municipal Police     Natrona   
290559     290560     WY01900    Sweetwater           Sheriff  Sweetwater   
290560     290561     WY01902  Rock Springs  Municipal Police  Sweetwater   
290561     290562     WY02101      Evanston  Municipal Police       Uinta   

          State  Year     Month  Incident              Crime Type  ...  \
266227   Alaska  1993       May         2  Murder or Manslaughter  ...   
266235   Alaska  1993       May         2  Murder or Manslaughter  ...   
266237   Alaska  1993       May         4  Murder or Manslaughter  ...   
266238   Alaska  1993       May         5  Murder or Manslaughter  ...   
266247   Alaska  1993   October         4  Murder or Manslaughter  ...   
...         ...   ...       ...       ...                     ...  ...   
290553  Wyoming  1993       May         1  Murder or Manslaughter  ...   
290555  Wyoming  1993  November         1  Murder or Manslaughter  ...   
290559  Wyoming  1993   January         1  Murder or Manslaughter  ...   
290560  Wyoming  1993     March         1  Murder or Manslaughter  ...   
290561  Wyoming  1993    August         1  Murder or Manslaughter  ...   

       Victim Ethnicity Perpetrator Sex  Perpetrator Age  \
266227          Unknown            Male               20   
266235          Unknown            Male               20   
266237          Unknown         Unknown                0   
266238          Unknown          Female               22   
266247          Unknown            Male               32   
...                 ...             ...              ...   
290553          Unknown            Male               47   
290555          Unknown            Male               17   
290559          Unknown            Male               28   
290560          Unknown          Female               20   
290561          Unknown            Male               21   

                     Perpetrator Race Perpetrator Ethnicity  Relationship  \
266227                          Black               Unknown        Sister   
266235                          Black               Unknown        Sister   
266237                        Unknown               Unknown       Unknown   
266238  Native American/Alaska Native               Unknown      Stranger   
266247                          Black               Unknown  Acquaintance   
...                               ...                   ...           ...   
290553                          White               Unknown        Family   
290555                          White               Unknown      Stranger   
290559                          White               Unknown      Stranger   
290560                          White               Unknown      Daughter   
290561                          White               Unknown       Ex-Wife   

              Weapon Victim Count Perpetrator Count Record Source  
266227       Handgun            1                 0           FBI  
266235       Handgun            1                 0           FBI  
266237       Handgun            0                 0           FBI  
266238       Unknown            0                 0           FBI  
266247         Knife            0                 0           FBI  
...              ...          ...               ...           ...  
290553       Handgun            0                 0           FBI  
2905

In [27]:
round ((data[(data['Year']==1993) & (data['Victim Sex']=='Female')].shape[0]  / data[data['Year']==1993].shape[0] *100) ,2)

21.98

21,98%  преступлений в "популярный" 1993 год точно совершены против женщин (есть точная информация). Мы ожидали, что этот процент будет больше

# Создание нового столбца и применение с ним фильтрации

In [28]:
data['Relationship'].unique()

array(['Acquaintance', 'Unknown', 'Wife', 'Stranger', 'Girlfriend',
       'Ex-Husband', 'Brother', 'Stepdaughter', 'Husband', 'Sister',
       'Friend', 'Family', 'Neighbor', 'Father', 'In-Law', 'Son',
       'Ex-Wife', 'Boyfriend', 'Mother', 'Common-Law Husband',
       'Common-Law Wife', 'Stepfather', 'Stepson', 'Stepmother',
       'Daughter', 'Boyfriend/Girlfriend', 'Employer', 'Employee'],
      dtype=object)

In [29]:
family_members = ['Ex-Husband', 'Wife', 'Brother', 'Stepdaughter', 'Husband', 'Sister', 'Family', 'Father', 'In-Law', 'Son','Ex-Wife', 'Mother', 'Common-Law Husband','Common-Law Wife', 'Stepfather', 'Stepson', 'Stepmother','Daughter']
data['Victim_is_Family_member']=data['Relationship'].apply(lambda x: 'Yes' if x in family_members else 'No')

Мы посмотрели, какие типы отношений существуют в таблице, выделили в список family_members тех, кто является членом семьи, далее создали столбец Victim_is_Family_member, где пропишем 'Yes', если жертва является для преступника членом семьи и 'No' в ином случае 

In [30]:
data[(data['Crime Solved']== 'Yes') & (data['Perpetrator Count']>1)]

Record ID Agency Code          Agency Name       Agency Type  \
57             58     AL00100            Jefferson           Sheriff   
169           170     AL00105           Gardendale  Municipal Police   
419           420     AL03900              Jackson           Sheriff   
423           424     AL04100           Lauderdale           Sheriff   
444           445     AL04400            Limestone           Sheriff   
...           ...         ...                  ...               ...   
638410     638411     WV03400      Nicholas County           Sheriff   
638421     638422     WV05000         Wayne County           Sheriff   
638434     638435     WVWSP55  State Police: Welch      State Police   
638435     638436     WVWSP55  State Police: Welch      State Police   
638444     638445     WY01101             Cheyenne  Municipal Police   

              City          State  Year      Month  Incident  \
57       Jefferson        Alabama  1980     August         3   
169      Jefferson        Alabama  1980       July         2   
419        Jackson        Alabama  1980    October         1   
423     Lauderdale        Alabama  1980   February         1   
444      Limestone        Alabama  1980  September         1   
...            ...            ...   ...        ...       ...   
638410    Nicholas  West Virginia  2014       June         1   
638421       Wayne  West Virginia  2014   November         1   
638434    McDowell  West Virginia  2014    October         1   
638435    McDowell  West Virginia  2014    October         1   
638444     Laramie        Wyoming  2014   December         1   

                    Crime Type  ... Perpetrator Sex Perpetrator Age  \
57      Murder or Manslaughter  ...            Male              22   
169     Murder or Manslaughter  ...            Male              32   
419     Murder or Manslaughter  ...            Male              32   
423     Murder or Manslaughter  ...          Female              22   
444     Murder or Manslaughter  ...            Male              33   
...                        ...  ...             ...             ...   
638410  Murder or Manslaughter  ...            Male              26   
638421  Murder or Manslaughter  ...            Male              20   
638434  Murder or Manslaughter  ...          Female              35   
638435  Murder or Manslaughter  ...          Female              35   
638444  Murder or Manslaughter  ...            Male              35   

        Perpetrator Race Perpetrator Ethnicity  Relationship        Weapon  \
57                 Black               Unknown      Stranger       Handgun   
169                White          Not Hispanic      Stranger       Handgun   
419                White               Unknown  Acquaintance  Blunt Object   
423                White          Not Hispanic  Acquaintance  Blunt Object   
444                White          Not Hispanic  Acquaintance       Shotgun   
...                  ...                   ...           ...           ...   
638410             White               Unknown  Acquaintance       Unknown   
638421             White               Unknown  Acquaintance         Knife   
638434             White               Unknown       Unknown       Firearm   
638435             White               Unknown       Unknown       Firearm   
638444             White               Unknown  Acquaintance   Suffocation   

        Victim Count Perpetrator Count Record Source Victim_is_Family_member  
57                 0                 2           FBI                      No  
169                0                 3           FBI                      No  
419                0                 4           FBI                      No  
423                0                 2           FBI                      No  
444                0                 5           FBI                      No  
...              ...               ...           ...                     ...  
638410             0           

Получается, что 24601 преступлений были раскрыты при условии, что преступников было больше одного. 
Теперь посмотрим, против кого чаще всего были соверешены преступления. Против членов семьи или посторонних людей? 
Обратимся к новому столбцу Victim_is_Family_member, который мы создали ранее. против кого чаще всего направлено преступление (против семьи или против незнакомцев - по новому столбцу)

In [31]:
data[(data['Crime Solved']== 'Yes') & (data['Perpetrator Count']>1) & (data['Victim_is_Family_member']=='Yes')]

Record ID Agency Code      Agency Name       Agency Type         City  \
2542         2543     CA01942      Los Angeles  Municipal Police  Los Angeles   
2690         2691     CA01942      Los Angeles  Municipal Police  Los Angeles   
2942         2943     CA01942      Los Angeles  Municipal Police  Los Angeles   
3082         3083     CA01942      Los Angeles  Municipal Police  Los Angeles   
5276         5277     FL00507        Melbourne  Municipal Police      Brevard   
...           ...         ...              ...               ...          ...   
635997     635998     TN06500    Morgan County           Sheriff       Morgan   
636527     636528     TX10100    Harris County           Sheriff       Harris   
637563     637564     UT01101       Cedar City  Municipal Police         Iron   
637755     637756     VA08300   Russell County           Sheriff      Russell   
638366     638367     WV00200  Berkeley County           Sheriff     Berkeley   

                State  Year     Month  Incident              Crime Type  ...  \
2542       California  1980      June        57  Murder or Manslaughter  ...   
2690       California  1980    August         7  Murder or Manslaughter  ...   
2942       California  1980   October        41  Murder or Manslaughter  ...   
3082       California  1980  December         8  Murder or Manslaughter  ...   
5276          Florida  1980     March         1  Murder or Manslaughter  ...   
...               ...   ...       ...       ...                     ...  ...   
635997      Tennessee  2014      June         1  Murder or Manslaughter  ...   
636527          Texas  2014       May         1  Murder or Manslaughter  ...   
637563           Utah  2014   October         1  Murder or Manslaughter  ...   
637755       Virginia  2014   January         2  Murder or Manslaughter  ...   
638366  West Virginia  2014   January         1  Murder or Manslaughter  ...   

       Perpetrator Sex Perpetrator Age               Perpetrator Race  \
2542            Female              23                          Black   
2690              Male              15                          Black   
2942            Female              33                          White   
3082              Male              28                          White   
5276            Female              36                          White   
...                ...             ...                            ...   
635997          Female              29                          White   
636527            Male              17                          White   
637563          Female              18  Native American/Alaska Native   
637755          Female              27                          White   
638366            Male              32                          White   

       Perpetrator Ethnicity        Relationship        Weapon  Victim Count  \
2542                 Unknown  Common-Law Husband       Shotgun             0   
2690            Not Hispanic             Brother       Firearm             0   
2942                Hispanic            Daughter  Blunt Object             0   
3082                Hispanic              In-Law         Knife             0   
5276            Not Hispanic             Husband  Blunt Object             0   
...                      ...                 ...           ...           ...   
635997               Unknown                 Son       Unknown             0   
636527          Not Hispanic          Stepfather  Blunt Object             0   
637563               Unknown          Stepfather         Rifle             0   
637755               Unknown             Brother       Unknown             0   
638366               Unknown                 Son  Blunt Object             0   

       Perpetrator Count Record Source Victim_is_Family_member  
2542                   3           FBI                     Yes  
2690                   2           FBI                     Yes  
2942                   2           FBI   

In [32]:
round((data[(data['Crime Solved']== 'Yes') & (data['Perpetrator Count']>1) & (data['Victim_is_Family_member']=='Yes')].shape[0]/
data[(data['Crime Solved']== 'Yes') & (data['Perpetrator Count']>1)].shape[0]*100),2)

2.74

In [33]:
data[(data['Crime Solved']== 'Yes') & (data['Perpetrator Count']>1) & (data['Victim_is_Family_member']=='Yes')]['Relationship'].value_counts()

Relationship
Family                181
Husband               109
Son                    66
Mother                 57
Father                 45
Wife                   35
Brother                34
Daughter               32
In-Law                 28
Stepfather             26
Sister                 20
Ex-Husband             14
Common-Law Husband      8
Stepmother              5
Stepson                 5
Common-Law Wife         4
Ex-Wife                 3
Stepdaughter            1
Name: count, dtype: int64

Почти 3% раскрытых преступлений (2,74),совершенных группой лиц больше 1, были направлены против членов семьи. При этом интересно, что чаще всего убивали видимо всю семью (значение Family намекает на это), на втором месте мужья 

# Сортировка

In [34]:
data.sort_values(['Year', 'Incident'], ascending=[True, False])

Record ID Agency Code      Agency Name       Agency Type        City  \
16293       16294     NY03030         New York  Municipal Police    New York   
16292       16293     NY03030         New York  Municipal Police    New York   
16291       16292     NY03030         New York  Municipal Police    New York   
16290       16291     NY03030         New York  Municipal Police    New York   
16289       16290     NY03030         New York  Municipal Police    New York   
...           ...         ...              ...               ...         ...   
638450     638451     WY01700  Sheridan County           Sheriff    Sheridan   
638451     638452     WY01701         Sheridan  Municipal Police    Sheridan   
638452     638453     WY01800  Sublette County           Sheriff    Sublette   
638453     638454     WY01902     Rock Springs  Municipal Police  Sweetwater   
627947     627948     FL05100     Pasco County           Sheriff       Pasco   

           State  Year      Month  Incident                  Crime Type  ...  \
16293   New York  1980   December       239  Manslaughter by Negligence  ...   
16292   New York  1980   December       238      Murder or Manslaughter  ...   
16291   New York  1980   December       237      Murder or Manslaughter  ...   
16290   New York  1980   December       236      Murder or Manslaughter  ...   
16289   New York  1980   December       235      Murder or Manslaughter  ...   
...          ...   ...        ...       ...                         ...  ...   
638450   Wyoming  2014       June         1      Murder or Manslaughter  ...   
638451   Wyoming  2014  September         1      Murder or Manslaughter  ...   
638452   Wyoming  2014   December         1      Murder or Manslaughter  ...   
638453   Wyoming  2014  September         1      Murder or Manslaughter  ...   
627947   Florida  2014  September         0      Murder or Manslaughter  ...   

       Perpetrator Sex Perpetrator Age        Perpetrator Race  \
16293             Male              21                   Black   
16292          Unknown               0                 Unknown   
16291          Unknown               0                 Unknown   
16290          Unknown               0                 Unknown   
16289          Unknown               0                 Unknown   
...                ...             ...                     ...   
638450            Male              57                   White   
638451          Female              22  Asian/Pacific Islander   
638452            Male              31                   White   
638453          Female              24                   White   
627947            Male               0                   White   

       Perpetrator Ethnicity  Relationship        Weapon  Victim Count  \
16293                Unknown        Friend       Handgun             0   
16292                Unknown       Unknown       Handgun             0   
16291                Unknown       Unknown       Handgun             0   
16290                Unknown       Unknown       Handgun             0   
16289                Unknown      Stranger       Handgun             0   
...                      ...           ...           ...           ...   
638450               Unknown  Acquaintance       Handgun             0   
638451               Unknown      Daughter   Suffocation             0   
638452          Not Hispanic      Stranger         Knife             0   
638453          Not Hispanic      Daughter  Blunt Object             0   
627947               Unknown       Unknown  Blunt Object             3   

       Perpetrator Count Record Source Victim_is_Family_member  
16293                  0           FBI                      No  
16292                  0           FBI                      No  
16291                  0           FBI                      No  
16290                  0           FBI                      No  
16289                  0           FBI                      No  
...           

К 1980 году в Нью-Йорке было совершено 239 преступлений (самый большой номер инцидента)

In [35]:
data.sort_values('Perpetrator Age', ascending=False)

Record ID Agency Code      Agency Name       Agency Type  \
284238     284239     OH04807           Toledo  Municipal Police   
284248     284249     OH04807           Toledo  Municipal Police   
417054     417055     WV02500           Marion           Sheriff   
634095     634096     OH02500  Franklin County           Sheriff   
421347     421348     FL04102        Bradenton  Municipal Police   
...           ...         ...              ...               ...   
288218     288219     TXDPD00           Dallas  Municipal Police   
288216     288217     TXDPD00           Dallas  Municipal Police   
288215     288216     TXDPD00           Dallas  Municipal Police   
288211     288212     TXDPD00           Dallas  Municipal Police   
319227     319228     DCMPD00       Washington  Municipal Police   

                        City                 State  Year     Month  Incident  \
284238                 Lucas                  Ohio  1993    August         1   
284248                 Lucas                  Ohio  1993   October         4   
417054                Marion         West Virginia  2000  November         1   
634095              Franklin                  Ohio  2014      July         1   
421347               Manatee               Florida  2001       May       425   
...                      ...                   ...   ...       ...       ...   
288218                Dallas                 Texas  1993   January        24   
288216                Dallas                 Texas  1993   January        22   
288215                Dallas                 Texas  1993   January        21   
288211                Dallas                 Texas  1993   January        17   
319227  District of Columbia  District of Columbia  1995   January        14   

                    Crime Type  ... Perpetrator Sex Perpetrator Age  \
284238  Murder or Manslaughter  ...         Unknown              99   
284248  Murder or Manslaughter  ...         Unknown              99   
417054  Murder or Manslaughter  ...            Male              99   
634095  Murder or Manslaughter  ...         Unknown              99   
421347  Murder or Manslaughter  ...         Unknown              99   
...                        ...  ...             ...             ...   
288218  Murder or Manslaughter  ...         Unknown               0   
288216  Murder or Manslaughter  ...         Unknown               0   
288215  Murder or Manslaughter  ...         Unknown               0   
288211  Murder or Manslaughter  ...         Unknown               0   
319227  Murder or Manslaughter  ...         Unknown               0   

        Perpetrator Race Perpetrator Ethnicity Relationship   Weapon  \
284238           Unknown               Unknown      Unknown  Handgun   
284248           Unknown               Unknown     Stranger  Handgun   
417054             White               Unknown     Stranger    Knife   
634095           Unknown               Unknown      Unknown  Firearm   
421347           Unknown               Unknown      Unknown    Knife   
...                  ...                   ...          ...      ...   
288218           Unknown               Unknown      Unknown  Handgun   
288216           Unknown               Unknown      Unknown  Firearm   
288215           Unknown               Unknown      Unknown  Firearm   
288211           Unknown               Unknown      Unknown  Handgun   
319227           Unknown               Unknown      Unknown  Handgun   

        Victim Count Perpetrator Count Record Source Victim_is_Family_member  
284238             0                 1           FBI                      No  
284248             0                 0           FBI                      No  
417054             0                 0           FBI                      No  
634095             0                 0           FBI                      No  
421347             0                 0          FOIA                      No  
...              ...               ...

При сортировке возраста преступника мы заметили, что есть странные возрасты 99, что также говорит о наличии иных пропусков в возрасте, кроме 0, который мы поставили изначальное

In [36]:
data.sort_values('Victim Count', ascending=False)

Record ID Agency Code    Agency Name       Agency Type         City  \
330101     330102     OK05506  Oklahoma City  Municipal Police     Oklahoma   
330157     330158     OK05506  Oklahoma City  Municipal Police     Oklahoma   
268686     268687     CA01942    Los Angeles  Municipal Police  Los Angeles   
330184     330185     OK05506  Oklahoma City  Municipal Police     Oklahoma   
330185     330186     OK05506  Oklahoma City  Municipal Police     Oklahoma   
...           ...         ...            ...               ...          ...   
220232     220233     AL05904      Alabaster  Municipal Police       Shelby   
220233     220234     AL05904      Alabaster  Municipal Police       Shelby   
220234     220235     AL06000         Sumter           Sheriff       Sumter   
220235     220236     AL06000         Sumter           Sheriff       Sumter   
638453     638454     WY01902   Rock Springs  Municipal Police   Sweetwater   

             State  Year      Month  Incident              Crime Type  ...  \
330101    Oklahoma  1995      April         9  Murder or Manslaughter  ...   
330157    Oklahoma  1995      April        19  Murder or Manslaughter  ...   
268686  California  1993    October         4  Murder or Manslaughter  ...   
330184    Oklahoma  1995      April         6  Murder or Manslaughter  ...   
330185    Oklahoma  1995      April         7  Murder or Manslaughter  ...   
...            ...   ...        ...       ...                     ...  ...   
220232     Alabama  1991    January         1  Murder or Manslaughter  ...   
220233     Alabama  1991       June         1  Murder or Manslaughter  ...   
220234     Alabama  1991      March         1  Murder or Manslaughter  ...   
220235     Alabama  1991   December         1  Murder or Manslaughter  ...   
638453     Wyoming  2014  September         1  Murder or Manslaughter  ...   

       Perpetrator Sex Perpetrator Age  Perpetrator Race  \
330101            Male              27             White   
330157            Male              27             White   
268686            Male              25             White   
330184            Male              27             White   
330185            Male              27             White   
...                ...             ...               ...   
220232            Male              30             Black   
220233            Male               0             Black   
220234         Unknown               0           Unknown   
220235            Male              40             Black   
638453          Female              24             White   

       Perpetrator Ethnicity     Relationship        Weapon  Victim Count  \
330101               Unknown         Stranger    Explosives            10   
330157               Unknown         Stranger    Explosives            10   
268686              Hispanic         Stranger          Fire            10   
330184               Unknown         Stranger    Explosives            10   
330185               Unknown         Stranger    Explosives            10   
...                      ...              ...           ...           ...   
220232               Unknown     Acquaintance       Handgun             0   
220233               Unknown           Family         Knife             0   
220234               Unknown          Unknown       Firearm             0   
220235               Unknown  Common-Law Wife       Handgun             0   
638453          Not Hispanic         Daughter  Blunt Object             0   

       Perpetrator Count Record Source Victim_is_Family_member  
330101                 2           FBI                      No  
330157                 2           FBI                      No  
268686                 1           FBI                      No  
330184                 2           FBI                      No  
330185                 2           FBI                      No  
...                  ...           ...                     ...  
220232               

Максимальное количество жертв равняется 10, при этом на них нападал 1 преступник или их было двое, видим также, что жертвы для преступников были незнакомцами. 

# СОЗДАНИЕ ЧАСТОТНОЙ ТАБЛИЦЫ

Частотные таблицы строятся, как правило на категорийных данных и показывают сколько раз уникальная категория встретилась в признаке.
Выведем наименования столбцов категорийных данных.

In [37]:
object_cols = data.select_dtypes('object').columns
object_cols

Index(['Agency Code', 'Agency Name', 'Agency Type', 'City', 'State', 'Month',
       'Crime Type', 'Crime Solved', 'Victim Sex', 'Victim Race',
       'Victim Ethnicity', 'Perpetrator Sex', 'Perpetrator Race',
       'Perpetrator Ethnicity', 'Relationship', 'Weapon', 'Record Source',
       'Victim_is_Family_member'],
      dtype='object')

Сформируем частотную таблицу наименования агенства ("Agency Name")

In [38]:
data['Agency Name'].value_counts()

Agency Name
New York                      38416
Los Angeles                   29007
Chicago                       21331
Detroit                       17206
Houston                       13046
                              ...  
Benedict College                  1
Riverside Township                1
White Haven Borough               1
South Londonderry Township        1
Sublette County                   1
Name: count, Length: 9216, dtype: int64

Узнаем какую долю занимает каждое агенство в общей таблице данных

In [39]:
data['Agency Name'].value_counts(normalize = True)

Agency Name
New York                      0.060170
Los Angeles                   0.045433
Chicago                       0.033410
Detroit                       0.026949
Houston                       0.020434
                                ...   
Benedict College              0.000002
Riverside Township            0.000002
White Haven Borough           0.000002
South Londonderry Township    0.000002
Sublette County               0.000002
Name: proportion, Length: 9216, dtype: float64

Построим частотную таблицу по данным наименование города "City" применив сортировку по убыванию. 

In [40]:
data['City'].value_counts().sort_values(ascending = False)

City
Los Angeles    44511
New York       38431
Cook           22383
Wayne          19904
Harris         16331
               ...  
Bowman             1
Rooks              1
McCone             1
Allamakee          1
Treasure           1
Name: count, Length: 1782, dtype: int64

Самый криминальный город- ГОРОД АНГЕЛОВ??? (Los Angeles)

А еще интересно насколько часто происходят преступления против членов свой семьи... 

In [41]:
data['Victim_is_Family_member'].value_counts(normalize = True)

Victim_is_Family_member
No     0.861309
Yes    0.138691
Name: proportion, dtype: float64

И мы видим что более чем 13% занимают преступления против своих близких.... 'Бытовуха'